tests/unit/TranslateTest.js

```diff
@@ -52,3 +54,42 @@ describe('translate', () => {
         expect(translate.translate('en', ['testKeyGroup', 'testFunction'], {testVariable})).toBe(expectedValue);
     });
 });
+
+describe('Translation Keys', () => {
+    function traverseKeyPath(source, path, keyPaths) {
+        const pathArray = keyPaths || [];
+        const keyPath = path ? `${path}.` : '';
+        _.each(_.keys(source), (key) => {
+            if (_.isObject(source[key]) && !_.isFunction(source[key])) {
+                traverseKeyPath(source[key], keyPath + key, pathArray);
+            } else {
+                pathArray.push(keyPath + key);
+            }
+        });
+        return pathArray;
+    }
+    const excludeLanguages = ['en', 'es-ES'];
+    const languages = _.without(_.keys(originalTranslations.default), ...excludeLanguages);
+    const mainLanguage = originalTranslations.default.en;
+    const mainLanguageKeys = traverseKeyPath(mainLanguage);
+
+    _.each(languages, (ln) => {
+        const languageKeys = traverseKeyPath(originalTranslations.default[ln]);
+
+        it(`Does ${ln} locale has all the keys`, () => {
+            const hasAllKeys = _.difference(mainLanguageKeys, languageKeys);
+            if (hasAllKeys.length) {
+                console.debug(`🏹 [ ${hasAllKeys.join(', ')} ] are missing from ${ln}.js`);
+            }
+            expect(hasAllKeys.length).toBe(0);
```

Comment:
So, I was thinking, maybe we could do this and get a better error message?
```suggestion
            expect(hasAllKeys).toBeEqual([]);
```
I think this will show in the test result exactly which keys are missing, no?
(same applies to the one below)

Reply:
It does not show the keys that are not correct. It just shows Expected and produced output. So all the correct and wrong keys would be shown as a big string which is like 50 lines long.

Reply:
I tested with `expect.objectContaining and expect.arrayContaining`.

Reply:
hmmmm, can we change the code to do `_.without(mainLagunageKeys, ...languageKeys)`? That would return only the missing keys, right?

Reply:
It currently only give the missing keys. Again `_.without(mainLagunageKeys, ...languageKeys)` will go in `console.log`. @iwiznia

Reply:
The console log is not good enough as it is super hard to find the right message. In any case I don't think you understood what I meant, because I just tried and it works. Here's the test code:
```
        it(`Does ${ln} locale has all the keys`, () => {
            const hasAllKeys = _.without(mainLanguageKeys, ...languageKeys);
            expect(hasAllKeys).toEqual([]);
        });

        it(`Does ${ln} locale has unused keys`, () => {
            const hasAllKeys = _.without(languageKeys, ...mainLanguageKeys);
            expect(hasAllKeys).toEqual([]);
        });
```
And this is the output:
```
 ● Translation Keys › Does es locale has all the keys

    expect(received).toEqual(expected) // deep equality

    - Expected  - 1
    + Received  + 4

    - Array []
    + Array [
    +   "preferencesPage.mostRecent",
    +   "passwordForm.error.incorrectLoginOrPassword",
    + ]

      79 |         it(`Does ${ln} locale has all the keys`, () => {
      80 |             const hasAllKeys = _.without(mainLanguageKeys, ...languageKeys);
    > 81 |             expect(hasAllKeys).toEqual([]);
         |                                ^
      82 |         });
      83 |
      84 |         it(`Does ${ln} locale has unused keys`, () => {

      at Object.<anonymous> (tests/unit/TranslateTest.js:81:32)

  ● Translation Keys › Does es locale has unused keys

    expect(received).toEqual(expected) // deep equality

    - Expected  - 1
    + Received  + 4

    - Array []
    + Array [
    +   "initialSettingsPage.appDownloadLinks.window.label",
    +   "initialSettingsPage.appDownloadLinks.Rajat.label",
    + ]

      84 |         it(`Does ${ln} locale has unused keys`, () => {
      85 |             const hasAllKeys = _.without(languageKeys, ...mainLanguageKeys);
    > 86 |             expect(hasAllKeys).toEqual([]);
         |                                ^
      87 |         });
      88 |     });
      89 | });

      at Object.<anonymous> (tests/unit/TranslateTest.js:86:32)

```

Reply:
oh Yeah. This way I got it now.

Reply:
@iwiznia  Check this out https://github.com/Expensify/App/actions/runs/1079637864. I added annotations.
![image](https://user-images.githubusercontent.com/24370807/127545308-f5290a18-16b7-4567-8555-1bb20fc5ab34.png)